Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -146,105 +265,104 @@ namespace AzAssetBrowserRequestHandlerPrivate
+                        targetEntityId, &EditorRequests::CreateNewEntityAtPosition, worldTransform.GetTranslation(), parentEntityId);
 
-        AZ::EntityId targetEntityId;
-        EditorRequests::Bus::BroadcastResult(targetEntityId, &EditorRequests::CreateNewEntityAtPosition, worldTransform.GetTranslation(), AZ::EntityId());
+                    AZ::Entity* newEntity = nullptr;
+                    AZ::ComponentApplicationBus::BroadcastResult(newEntity, &AZ::ComponentApplicationRequests::FindEntity, targetEntityId);
 
-        AZ::Entity* newEntity = nullptr;
-        AZ::ComponentApplicationBus::BroadcastResult(newEntity, &AZ::ComponentApplicationRequests::FindEntity, targetEntityId);
+                    if (newEntity == nullptr)
+                    {
+                        QMessageBox::warning(
+                            mainWindow,
+                            QObject::tr("Asset Drop Failed"),
+                            QStringLiteral("Could not create entity from selected asset(s)."));
+                        return;
+                    }
 
-        if (newEntity == nullptr)
-        {
-            QMessageBox::warning(mainWindow, QObject::tr("Asset Drop Failed"), QStringLiteral("Could not create entity from selected asset(s)."));
-            return;
-        }
+                    // Deactivate the entity so the properties on the components can be set.
+                    newEntity->Deactivate();
 
-        // Deactivate the entity so the properties on the components can be set.
-        newEntity->Deactivate();
+                    newEntity->SetName(entityName);
 
-        newEntity->SetName(entityName);
+                    AZ::ComponentTypeList componentsToAdd;
+                    componentsToAdd.push_back(componentTypeId);
 
+                     // Add the product as components to this entity.
+                    AZStd::vector<AZ::EntityId> entityIds = { targetEntityId };
+                    EntityCompositionRequests::AddComponentsOutcome addComponentsOutcome = AZ::Failure(AZStd::string());
+                    EntityCompositionRequestBus::BroadcastResult(
+                        addComponentsOutcome, &EntityCompositionRequests::AddComponentsToEntities, entityIds, componentsToAdd);
+                    
+                    if (!addComponentsOutcome.IsSuccess())
+                    {
+                        AZ_Error(
+                            "AssetBrowser",
+                            false,
+                            "Could not create the requested components from the selected assets: %s",
+                            addComponentsOutcome.GetError().c_str());
+                        EditorEntityContextRequestBus::Broadcast(&EditorEntityContextRequests::DestroyEditorEntity, targetEntityId);
+                        return;
+                    }
+                    // activate the entity first, so that the primary asset change is done in the context of it being awake.
```

Comment:
lawsonamzn note:  This alone fixes a plethora of bugs related to the asset setting not working or crashing.